### PR TITLE
Add script 2 for cell-contamination-removal-analysis module

### LINKED ISSUE #98: Proposed Analysis: To add script 2 for cell-contamination-removal-analysis module
## What are the scientific goals of the analysis?

To add the script for the integrative analysis in the `cell-contamination-removal-analysis` module, i.e., `02-integrative-analysis.Rmd`.

### PR DESCRIPTION
**Pull request by AntoniaChroni**
# Purpose/implementation Section

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## What scientific question/module of analysis is your analysis addressing?

To add the script for the integrative analysis in the `cell-contamination-removal-analysis` module, i.e., `02-integrative-analysis.Rmd`.

## What GitHub issue does your pull request address?

Closes #98.

## Directions for reviewers 
Tell potential reviewers what kind of feedback you are soliciting.

### Which areas should receive a particularly close look?

Please review for documentation and code logic and clarity.

## Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.

## Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

File: analyses/cell-contamination-removal-analysis/02-integrative-analysis.Rmd [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+(2) [Harmony](https://portals.broadinstitute.org/harmony/) integration workflow by [Korsunsky et al., 2019](https://www.nature.com/articles/s41592-019-0619-0), and
+(3) [Liger](https://welch-lab.github.io/liger/articles/liger_with_seurat.html) integration workflow by [Welch et al., 2019](https://doi.org/10.1016/j.cell.2019.05.006).
+
+Different number of dimensions for different integration algorithms (Seurat, Harmony, Liger, etc.) is common and depending on the algorithm they come with their corresponding variables depending on how the algorithm is being used. Each of these integration algorithm takes into account typically the same variables but the math they use to transform them is so different that you can't use the same number of resolutions. Even though they use the same word, resolution doesn't actually effect each algorithm the same way and their maximums can be different. 30 for Seurat, 30 for Harmony and 20 for Liger give the most similar integration results when all other variables are taken into account while also not wasting too much compute time.
```
Comment (sharonfreshour) [inline analyses/cell-contamination-removal-analysis/02-integrative-analysis.Rmd:150]: I think this report should be updated to match the report generated in the `intergrative-analysis` module for clarity and consistency.

```suggestion
Different integration algorithms, such as Seurat, Harmony, and Liger, often use different numbers of dimensions. The optimal number of dimensions depends on the specific algorithm and its associated parameters. While these algorithms generally consider similar variables, the mathematical approaches they use to process and transform those variables differ significantly, which means the same number of dimensions (or "resolutions") doesn't work equally well across all algorithms.

Even though they all refer to "resolution," the effect of resolution can vary from one algorithm to another, and the maximum number of dimensions they can effectively use may differ. Based on our experience, we recommend using 30 dimensions for Seurat, 30 for Harmony, and 20 for Liger. These settings tend to produce the most consistent integration results without unnecessarily increasing computational load.
```

**Review by sharonfreshour — COMMENTED** [review]
The code looks good and seems logical. I suggested to updated the explanation of dimensions being used for integration to improve clarity and maintain consistency across modules.

File: analyses/cell-contamination-removal-analysis/02-integrative-analysis.Rmd (hunk 1 above)
Comment (AntoniaChroni) [reply analyses/cell-contamination-removal-analysis/02-integrative-analysis.Rmd]: Thank you for noticing this! The notebook has been updated here 7acd646a58e08fdcd2456571df2425774eaf0ead.

**Review by sharonfreshour — APPROVED** [review]
Updates look good and match report for `integrative-analysis` module.

**Pull request merged**